### PR TITLE
Support YAML configuration settings

### PR DESCRIPTION
**Pull request by cschwede**

### CODE REVIEW

File: pkg/controller/embercsi/common.go [hunk 1]
```diff
@@ -263,7 +267,8 @@ func generateVolumeMounts(ecsi *embercsiv1alpha1.EmberCSI, csiDriverMode string)
 	}
 
 	// Check to see if the volume driver is LVM
-	if len(ecsi.Spec.Config.EnvVars.X_CSI_BACKEND_CONFIG) > 0 && strings.Contains(strings.ToLower(ecsi.Spec.Config.EnvVars.X_CSI_BACKEND_CONFIG), "lvmvolume") {
+	X_CSI_BACKEND_CONFIG := interfaceToString(ecsi.Spec.Config.EnvVars.X_CSI_BACKEND_CONFIG)
+	if len(X_CSI_BACKEND_CONFIG) > 0 && strings.Contains(strings.ToLower(X_CSI_BACKEND_CONFIG), "lvmvolume") {
```
Comment (Akrog) [inline pkg/controller/embercsi/common.go]: Isn't this kind of broad? This should be checking just the "driver" part, right? Not the whole json blob.

File: pkg/controller/embercsi/common.go [hunk 2]
```diff
@@ -383,7 +388,8 @@ func generateVolumes(ecsi *embercsiv1alpha1.EmberCSI, csiDriverMode string) []co
...
```
Comment (Akrog) [inline pkg/controller/embercsi/common.go]: ditto

File: pkg/controller/embercsi/common.go (hunk 1 above)
Comment (cschwede) [reply pkg/controller/embercsi/common.go]: Yes, fully agree, and it's on my list for future patches :)

**Comment by cschwede** [comment]
Merged after two positive comments/approvals.

**Pull request merged**